Добавлен интерфейс для таблицы Orders.

mrnchernova opened the pull request:
Добавлен интерфейс для таблицы Orders.
Приведен пример работы с заказами при помощи метода getById

Commit:
Добавлен интерфейс для таблицы Orders.
Приведен пример работы с заказами при помощи метода getById

RomanKashenok commented on src/main/java/by/trucking/model/Client.java:
```diff
@@ -0,0 +1,46 @@
+public class Client {
+
+    private int id;
+    private String title;
```
Тайтл клиента - как минимум странно...

RomanKashenok commented on src/main/java/by/trucking/model/Client.java:
```diff
@@ -0,0 +1,46 @@
+
+    private int id;
+    private String title;
+    private String contact;
```
Контакт скорее всего должен быть либо мапой (улица - ..., дом - ... и т.д.) либо классом Contact

RomanKashenok replied:
что с этим?

RomanKashenok commented on src/main/java/by/trucking/model/Client.java:
```diff
@@ -0,0 +1,46 @@
+    private String contact;
+    private User user;
+
+    public Client() {
+    }
+
+    public Client(int id, String title, String contact) {
+        this.id = id;
+        this.title = title;
+        this.contact = contact;
+    }
+
+    public int getId() {
+        return id;
+    }
+
+    public void setId(int id) {
+        this.id = id;
+    }
+
+    public String getTitle() {
+        return title;
+    }
+
+    public void setTitle(String title) {
+        this.title = title;
+    }
+
+    public String getContact() {
+        return contact;
+    }
+
+    public void setContact(String contact) {
+        this.contact = contact;
+    }
+
+    public User getUser() {
+        return user;
+    }
```
давай это все под lombok спрячем

RomanKashenok commented on src/main/java/by/trucking/model/Driver.java:
```diff
@@ -0,0 +1,31 @@
+    private Organization organization;
+    private Status status;
+
+    public Driver() {
+    }
+
+    public Driver(int id, String name, String surname) {
+        this.id = id;
+        this.name = name;
+        this.surname = surname;
+    }
+
+
+    public Organization getOrganization() {
+        return organization;
+    }
+
+    public Status getStatus() {
+        return status;
+    }
+
+    public void setStatus(Status status) {
+        this.status = status;
+    }
```
ломбок?

RomanKashenok commented on src/main/java/by/trucking/model/Order.java:
```diff
@@ -0,0 +1,98 @@
+
+public class Order {
+    private int id;
+    private String cargo;
```
я все равно не понимаю что такое тут карго...

RomanKashenok commented on src/main/java/by/trucking/model/Organization.java:
```diff
@@ -0,0 +1,35 @@
+
+    private int id;
+    private String title;
+    private User user;
```
наверное тут бы еще список машин и список водителей?
и еще список всех заказов

RomanKashenok replied:
ну и адрес наверное и какие там еще данные

RomanKashenok commented on src/main/java/by/trucking/model/StatusOrder.java:
```diff
@@ -0,0 +1,28 @@
+
+public class StatusOrder {
+    private int id;
+    private String status;
```
это наверное и енамом можно сделать

RomanKashenok commented on src/main/java/by/trucking/model/Status.java:
```diff
@@ -0,0 +1,28 @@
+package by.trucking.model;
+
+public class Status {
```
а это статус чего?

RomanKashenok commented on src/main/java/by/trucking/model/Role.java:
```diff
@@ -0,0 +1,27 @@
+package by.trucking.model;
+
+public class Role {
```
можно енамом сделать

RomanKashenok commented on src/main/java/by/trucking/model/Transport.java:
```diff
@@ -0,0 +1,73 @@
+public class Transport {
+    private int id;
+    private String model;
+    private String vehicle_type;
```
енам?

RomanKashenok commented on src/main/java/by/trucking/model/Transport.java:
```diff
@@ -0,0 +1,73 @@
+    private String carrying_capacity;
+    private String number_plate;
+    private Status status;
+    private Organization organization;
```
скорее в организации числится транспорт, а не транспорт имеет организацию

RomanKashenok commented on src/main/java/by/trucking/repository/OrderRepositoryDBImpl.java:
```diff
@@ -0,0 +1,94 @@
+                            rs.getFloat(6));
+
+                } else {
+                    return new Order();
```
налл давай возвращать

RomanKashenok commented on src/main/java/by/trucking/service/ConnectionDB.java:
```diff
@@ -0,0 +1,22 @@
+
+    public static Connection connection;
+
+    static {
```
нене, давайте не будемте статик инициализацию юзать

Commit:
Класс Status теперь Enum
Добавлен класс VehicleType (тип транспорта). Тоже enum.
Добавлена библиотека lombok

Commit:
Класс role  изменен на enum; добавлена зависимость javax.servlet

Commit:
servlet class

Commit:
tomcat

Commit:
getOrders method

Commit:
tomcat

Commit:
tomcat

Commit:
Merge remote-tracking branch 'origin/Interfaces_w' into Interfaces_w

Commit:
tomcat

Commit:
tomcat

Commit:
выводит из БД в браузер (проблема с кодировкой)

Commit:
Выводит все заказы в браузер

mrnchernova committed:
Delete .idea directory

mrnchernova committed:
Update .gitignore

mrnchernova committed:
Update .gitignore

mrnchernova committed:
Delete part_2.iml

RomanKashenok committed:
interfaces branch comment

Commit:
bug fixing

mrnchernova committed:
Update README.md

mrnchernova committed:
Delete target/part_2-1.0-SNAPSHOT directory

Commit:
Выводит все заказы и всех клиентов

RomanKashenok commented on pom.xml:
```diff
@@ -41,7 +41,12 @@
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-war-plugin</artifactId>
+            <version>3.3.2</version>
+        </dependency>
```
убирай отсюда. это в билд секции должно быть

RomanKashenok commented on src/main/java/by/trucking/Runner.java:
```diff
@@ -1,22 +1,33 @@
 
 public class Runner {
     public static void main(String[] args) throws SQLException {
 
```
давайте уже убирать метод main и классы, его содержащие

RomanKashenok commented on src/main/java/by/trucking/controller/ClientController.java:
```diff
@@ -0,0 +1,48 @@
+import java.io.PrintWriter;
+import java.util.List;
+
+@WebServlet("/getclients")
```
просто /clients

RomanKashenok commented on src/main/java/by/trucking/controller/OrderController.java:
```diff
@@ -0,0 +1,62 @@
+import java.sql.SQLException;
+import java.util.List;
+
+@WebServlet("/getorders")
```
просто /orders

RomanKashenok commented on src/main/java/by/trucking/controller/OrderController.java:
```diff
@@ -0,0 +1,62 @@
+                writer.println(orders.get(i).getWeight());
+                writer.println(orders.get(i).getDeparture());
+                writer.println(orders.get(i).getDestination());
+                writer.println(cs.getById(orders.get(i).getClient().getId()).getTitle());
```
нужно сначала все данные собрать, а потом выводить

RomanKashenok commented on src/main/java/by/trucking/controller/OrderController.java:
```diff
@@ -0,0 +1,62 @@
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        } finally {
+            writer.close();
```
помести в ресурсы блока try

RomanKashenok commented on src/main/java/by/trucking/controller/OrderController.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
```
убирай, если не используешь

RomanKashenok commented on src/main/java/by/trucking/repository/ClientRepositoryDBImpl.java:
```diff
@@ -0,0 +1,75 @@
+                            new User(rs.getInt(4)));
+
+                } else {
+                    return new Client();
```
return null

RomanKashenok commented on src/main/java/by/trucking/repository/UserRepositoryImpl.java:
```diff
@@ -0,0 +1,80 @@
+                            Role.getByOrdinal(rs.getInt(4)));
+
+                } else {
+                    return new User();
```
return null

RomanKashenok commented on src/main/java/by/trucking/service/ClientServiceImpl.java:
```diff
@@ -0,0 +1,52 @@
+
+    @Override
+    public List<Client> getClients() {
+        List<Client> clients = clientRepository.getClients();
```
возвращай сразу

Commit:
corrected some mistakes

Commit:
Исправлены замечания; добавлен класс Contact (int unp,String phone) для дальнейшей его сериализации и добавления полученных данных в таблицы Organizations и Clients

RomanKashenok commented on src/main/java/by/trucking/repository/OrderRepository.java:
```diff
@@ -17,11 +17,11 @@ public interface OrderRepository {
     List<Order> getOrders();
 
-    List<Order> getByCargo(String cargo);
+    List<Order> getByDescription(String description);
```
remove

RomanKashenok commented on src/main/java/by/trucking/service/OrderService.java:
```diff
@@ -18,7 +18,7 @@ public interface OrderService {
     List<Order> getOrders();
 
-    List<Order> getByCargo(String cargo);
+    List<Order> getByDescription(String description);
```
remove

RomanKashenok commented on src/main/java/by/trucking/service/OrderServiceImpl.java:
```diff
@@ -51,8 +51,8 @@ public Order getById(int id) throws SQLException {
-    public List<Order> getByCargo(String cargo) {
-        return orderRepository.getByCargo(cargo);
+    public List<Order> getByDescription(String description) {
+        return orderRepository.getByDescription(description);
```
remove

RomanKashenok reviewed (approved):
remove what needs to be removed

Commit:
remove getByDescription()

mrnchernova committed:
Merge branch 'main' into Interfaces

Commit:
resolved conflict
